fe/fe-core/src/main/java/com/starrocks/common/ThriftServer.java

```diff
@@ -143,9 +143,15 @@ public void start() throws IOException {
         serverThread = new Thread(new Runnable() {
             @Override
             public void run() {
-                server.serve();
+                try {
+                    server.serve();
+                } catch (Throwable t) {
+                    LOG.error("thrift server accept daemon failed, will exit", t);
+                    System.exit(-1);
```

Comment:
The `System.exit()` should be avoided to call even if the case of emergency.
Instead, we should return an error.